src/pieces/copilot/pieces_ask_websocket.py

```diff
@@ -53,32 +50,27 @@ def on_message(self,ws, message):
                 if self.verbose:
                     self.live.update(Markdown(self.final_answer), refresh=True)
                     self.live.stop()
- 
 
                 self.conversation = response.conversation
+                self.message_completed.set()
 
-                self.message_compeleted.set()
+            if self.on_message_callback:
+                self.on_message_callback(response)
 
         except Exception as e:
             print(f"Error processing message: {e}")
 
     def on_error(self, ws, error):
-        """Handle websocket errors."""
-        print(f"WebSocket error: {error}")
-        self.is_connected = False
+        print(f"AskWebsocket Error: {error}")
+        super().on_error(ws, error)
 
     def on_close(self, ws, close_status_code, close_msg):
-        """Handle websocket closure."""
-        # if self.verbose:
-        #     print("WebSocket closed")
-        self.is_connected = False
+        print(f"AskWebsocket closed. Status code: {close_status_code}, Message: {close_msg}")
```

Comment:
No need for this too

Reply:
addressed in commit [73566e9](https://github.com/pieces-app/cli-agent/pull/167/commits/73566e9e9375409f4ec01b4207affc4ea7cec9f2) commit